include/verilated_vpi.cpp

```diff
@@ -494,10 +499,14 @@ class VerilatedVpiImp {
         bool called = false;
         typedef std::set<VerilatedVpioVar*> VpioVarSet;
         VpioVarSet update;  // set of objects to update after callbacks
-        const auto end = cbObjList.end();  // prevent looping over newly added elements
-        for (auto it = cbObjList.begin(); it != end;) {
+        if (cbObjList.empty()) { return called; }
+        const auto last = std::prev(cbObjList.end());  // prevent looping over newly added elements
+        bool is_last;
+        for (auto it = cbObjList.begin();;) {
+            is_last = it == last;
```

Comment:
```suggestion
        for (auto it = cbObjList.begin(); true;) {
            // cbReasonRemove sets to nullptr, so we know on removal the old end() will still exist
            bool was_last = it == last;
```

Think "was_last" better indicates intent. Similar above as per codacy.